### PR TITLE
Forums tab 3

### PR DESCRIPTION
**Pull request by Sukhamjot-Singh**

### CODE REVIEW

**Comment by Sukhamjot-Singh** [comment]
Okay, so I tried calling [markRead](https://github.com/RetroShare/libretroshare/blob/3b829576b0b5333aace580d58f64234e3568ecb1/src/retroshare/rsgxsforums.h#L320) with an object which has groupId as key and msgID as value and good news is it did not crash.
image

BUT, its always returning false :(

Can you confirm if it is `mGroupId, mMsgId`.
This is what I am passing. 
`{"messageId":{"3e321965a3beb9cf4da43a29a2ef0f89":"28189c5c0402c29842a87e96804e1ff28363f197"},"read":false}`

**Comment by Sukhamjot-Singh** [comment]
> Okay, so I tried calling [markRead](https://github.com/RetroShare/libretroshare/blob/3b829576b0b5333aace580d58f64234e3568ecb1/src/retroshare/rsgxsforums.h#L320) with an object which has groupId as key and msgID as value and good news is it did not crash. image
> 
> BUT, its always returning false :(
> 
> Can you confirm if it is `mGroupId, mMsgId`. This is what I am passing. `{"messageId":{"3e321965a3beb9cf4da43a29a2ef0f89":"28189c5c0402c29842a87e96804e1ff28363f197"},"read":false}`
> 
> @csoler @G10h4ck
2. Can anyone also point me towards the API for getting the current Logged in ID.(The identity we are currently logged in by.)
cc: @csoler @G10h4ck

**Comment by csoler** [comment]
So did you manage to do it eventually? I am not sure how to interpret your last commit

**Comment by Sukhamjot-Singh** [comment]
> So did you manage to do it eventually? I am not sure how to interpret your last commit

Yes, G104hck suggested to use {first: , second:}, it worked perfectly

**Comment by Sukhamjot-Singh** [comment]
I am not sure but I can't seem to get `rsPeers/getOwnId `or `rsPeers/getGPGOwnId` get working. Maybe something on my side but can you check @csoler ?

**Comment by csoler** [comment]
I'm 100% sure they work (I mean the C++ methods). But you need to have started the core of course (meaning have logged in) for them to reply with a correct answer.

**Comment by Sukhamjot-Singh** [comment]
image

What exactly are these messages? This is a screenshot of a thread in Forum.

**Comment by csoler** [comment]
> image
> 
> What exactly are these messages? This is a screenshot of a thread in Forum.

That means the message is missing because none of your friends have it. That happens e.g. when the writer of the message has been banned by too many people in the close network which prevented RS to find a route to send the message to your own node.

**Comment by defnax** [comment]
sorry i was away to test your changes...better comments view will done too? like how on youtube? i saw last you not continoued on it, needs for channel and board commenting

**Comment by defnax** [comment]
this needs to be merged again to much changes :)

**Comment by defnax** [comment]
hi @Sukhamjot-Singh 
possible this feature is missing? for create forum/channel if public, circle restricted or node restricted
image

i get error on boards creation, not checked if your code is ready for creation
image

**Comment by defnax** [comment]
hi you has not added yet this:
image
but i dont know on boards i get a error output, im not sure how you ignore  the **circleType** value

image

**Comment by Sukhamjot-Singh** [comment]
> hi you has not added yet this: image but i dont know on boards i get a error output, im not sure how you ignore the **circleType** value
> 
> image
> 
> but the test boards is created and visible on the qt gui.

This is because in the channels API, the circle type is optional

image

In rsposted API, the circle type and id is required.
I will add this feature soon.

**Comment by csoler** [comment]
it's probably time to merge this in?

**Comment by Sukhamjot-Singh** [comment]
> it's probably time to merge this in?

Yes, it is stable to merge.

**Comment by Sukhamjot-Singh** [comment]
image

I can't seem to get this working.
Original code from @rottencandy looks like this 

image

Maybe it is something to do with the callback as I think the syntax is right.

**Comment by defnax** [comment]
why not merge? better for testing when its merged, you continoue again :)

**Comment by defnax** [comment]
@Sukhamjot-Singh remove the **draft** from the PR :)

**Comment by defnax** [comment]
Hi the creator window its on my screen on bottom and small window
image

can you center it and make it bigger? at the moment to much on the bottom

**Comment by Sukhamjot-Singh** [comment]
> Hi the creator window its on my screen on bottom and small window
> image
> 
> can you center it and make it bigger? at the moment to much on the bottom, the last parts on creator window not visible on my notebook.
> 

Oh Okay
I will look into it.

**Comment by defnax** [comment]
you latest update:
image

File: webui-src/app/forums/forums_util.js [hunk 1]
```diff
@@ -148,6 +171,23 @@ const SearchBar = () => {
       }),
   };
 };
+function popupmessage(message) {
```
Comment (rottencandy) [inline webui-src/app/forums/forums_util.js:174]: this is already in widgets.js too

File: webui-src/app/files/files_util.js [hunk 2]
```diff
@@ -82,9 +85,26 @@ function fileAction(hash, action) {
       console.error('Unknown action in Downloads.control()');
       return;
   }
-  rs.rsJsonApiRequest(actionHeader, jsonParams, () => {});
+  const res = await rs.rsJsonApiRequest(actionHeader, jsonParams, () => {});
+  return res.body.retval;
+}
+function popupmessage(message) {
```
[the remark this answers is not in the document]
Comment (Sukhamjot-Singh) [reply webui-src/app/files/files_util.js:91]: I will reuse this component wherever required now.

File: webui-src/app/files/friends_files.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                  haveFile
+                    ? ''
+                    : (await rs.rsJsonApiRequest('/rsFiles/FileRequest', {
+                        fileName: parStruct.details.name,
+                        hash: parStruct.details.hash,
+                        flags: util.RS_FILE_REQ_ANONYMOUS_ROUTING,
+                        size: {
+                          xstr64: parStruct.details.size.xstr64,
+                        },
+                      })) && m.redraw();
```
[the remark this answers is not in the document]
Comment (Sukhamjot-Singh) [reply webui-src/app/files/friends_files.js:79]: No. Although I will add a new dialog box to confirm start downloading.

File: webui-src/app/files/friends_files.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+              })
+            )
+          : m('td', ''),
+        m(
+          'td',
+          {
+            style: {
+              position: 'relative',
+              '--replyDepth': v.attrs.replyDepth,
+              left: 'calc(30px*var(--replyDepth))',
```
[the remark this answers is not in the document]
Comment (Sukhamjot-Singh) [reply webui-src/app/files/friends_files.js:56]: Yes, we can

**Comment by defnax** [comment]
@csoler please merg this

**Comment by csoler** [comment]
@rottencandy: are we good to merge?

**Comment by rottencandy** [comment]
Yes, I think we can merge this.

**Pull request merged**